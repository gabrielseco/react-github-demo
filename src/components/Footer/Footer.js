// @flow
import React from 'react';
import styles from './Footer.scss';
import applyClasses from './../../utils/applyClasses';

const footerIconClassNames = applyClasses({
  footerIcon: true,
  'fa fa-heart-o': true
});

const Footer = () => (
  <footer className={styles.footer}>
    <p>Gabriel Garc√≠a Seco</p>
    <p>
      Made with <i className={footerIconClassNames} />, Madrid 2018
    </p>
  </footer>
);

export default Footer;
