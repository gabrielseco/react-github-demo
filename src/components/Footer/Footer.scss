$footer-background-color: #212240;
$footer-text-color: #ffffff;
$footer-height: 4.6875em;
$footer-padding: 0.8em;

.footer {
  align-items: center;
  background: $footer-background-color;
  color: $footer-text-color;
  display: flex;
  height: $footer-height;
  justify-content: space-between;
  opacity: 0.9;
  padding: $footer-padding;
}

.footerIcon {
  margin: 0 4px;
}
