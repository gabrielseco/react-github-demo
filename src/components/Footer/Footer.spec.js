import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Footer from './Footer';

describe('Footer suite', () => {
  it('should render the Footer component', () => {
    const component = shallow(<Footer />);
    const tree = renderer.create(<Footer />).toJSON();

    expect(component).toBeDefined();
    expect(tree).toMatchSnapshot();
  });
});
