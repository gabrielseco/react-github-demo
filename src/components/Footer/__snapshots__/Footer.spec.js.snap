// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer suite should render the Footer component 1`] = `
<footer
  className="footer"
>
  <p>
    Gabriel Garc√≠a Seco
  </p>
  <p>
    Made with 
    <i
      className="footerIcon fa fa-heart-o"
    />
    , Madrid 2018 
  </p>
</footer>
`;
